



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="A tutorial covering practical elements of metagenomic analysis">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.4.3">
    
    
      
        <title>Hi-C - Metagenomics Tutorial</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.30686662.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.a8b3c06d.css">
      
      
        
        
        <meta name="theme-color" content="#3f51b5">
      
    
    
      <script src="../assets/javascripts/modernizr.74668098.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../css/extra.css">
    
    
      
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#exploiting-dna-dna-proximity" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href=".." title="Metagenomics Tutorial" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Metagenomics Tutorial
            </span>
            <span class="md-header-nav__topic">
              
                Hi-C
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/thegeneschool/metagenomics/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    thegeneschool/metagenomics
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
        

<nav class="md-tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    </ul>
  </div>
</nav>
      
      <main class="md-main" role="main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href=".." title="Metagenomics Tutorial" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Metagenomics Tutorial
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/thegeneschool/metagenomics/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    thegeneschool/metagenomics
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../startup/" title="Jupyter Setup" class="md-nav__link">
      Jupyter Setup
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../qc/" title="Quality Control" class="md-nav__link">
      Quality Control
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../assembly/" title="Assembly" class="md-nav__link">
      Assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../binning/" title="Binning" class="md-nav__link">
      Binning
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Hi-C
      </label>
    
    <a href="./" title="Hi-C" class="md-nav__link md-nav__link--active">
      Hi-C
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#what-is-metagenomic-hi-c" class="md-nav__link">
    What is Metagenomic Hi-C?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-hi-c-library-protocol" class="md-nav__link">
    The Hi-C Library Protocol
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#basic-concept" class="md-nav__link">
    Basic Concept
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#some-commercial-companies-offering-varieties-of-hi-c-kits-and-services" class="md-nav__link">
    Some commercial companies offering varieties of Hi-C kits and services
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#rough-protocol-outline" class="md-nav__link">
    Rough Protocol Outline
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-control-is-my-library-ok" class="md-nav__link">
    Quality Control: is my library ok?
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#evidence-of-proximity-ligation" class="md-nav__link">
    Evidence of Proximity Ligation
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#long-range-pairs" class="md-nav__link">
    Long-range pairs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#proximity-ligation-junctions" class="md-nav__link">
    Proximity ligation junctions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-a-qc-tool" class="md-nav__link">
    Using a QC Tool
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#get-some-hi-c-data-and-tools" class="md-nav__link">
    Get some Hi-C data and tools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#create-a-metagenome-assembly-and-map-hi-c-reads" class="md-nav__link">
    Create a metagenome assembly and map Hi-C reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bam-mode-analysis-with-qc3c" class="md-nav__link">
    BAM mode analysis with qc3C
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#estimation-of-fragment-size" class="md-nav__link">
    Estimation of fragment size
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#looking-at-the-results" class="md-nav__link">
    Looking at the results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#k-mer-mode-analysis-with-qc3c" class="md-nav__link">
    k-mer mode analysis with qc3C
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#looking-at-the-results_1" class="md-nav__link">
    Looking at the results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#metagenome-binning-with-hi-c-data" class="md-nav__link">
    Metagenome Binning with Hi-C Data
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#looking-at-the-results_2" class="md-nav__link">
    Looking at the results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../anvio/" title="Visualization" class="md-nav__link">
      Visualization
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../everythingelse/" title="Everything Else" class="md-nav__link">
      Everything Else
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#what-is-metagenomic-hi-c" class="md-nav__link">
    What is Metagenomic Hi-C?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-hi-c-library-protocol" class="md-nav__link">
    The Hi-C Library Protocol
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#basic-concept" class="md-nav__link">
    Basic Concept
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#some-commercial-companies-offering-varieties-of-hi-c-kits-and-services" class="md-nav__link">
    Some commercial companies offering varieties of Hi-C kits and services
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#rough-protocol-outline" class="md-nav__link">
    Rough Protocol Outline
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-control-is-my-library-ok" class="md-nav__link">
    Quality Control: is my library ok?
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#evidence-of-proximity-ligation" class="md-nav__link">
    Evidence of Proximity Ligation
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#long-range-pairs" class="md-nav__link">
    Long-range pairs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#proximity-ligation-junctions" class="md-nav__link">
    Proximity ligation junctions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-a-qc-tool" class="md-nav__link">
    Using a QC Tool
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#get-some-hi-c-data-and-tools" class="md-nav__link">
    Get some Hi-C data and tools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#create-a-metagenome-assembly-and-map-hi-c-reads" class="md-nav__link">
    Create a metagenome assembly and map Hi-C reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bam-mode-analysis-with-qc3c" class="md-nav__link">
    BAM mode analysis with qc3C
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#estimation-of-fragment-size" class="md-nav__link">
    Estimation of fragment size
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#looking-at-the-results" class="md-nav__link">
    Looking at the results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#k-mer-mode-analysis-with-qc3c" class="md-nav__link">
    k-mer mode analysis with qc3C
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#looking-at-the-results_1" class="md-nav__link">
    Looking at the results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#metagenome-binning-with-hi-c-data" class="md-nav__link">
    Metagenome Binning with Hi-C Data
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#looking-at-the-results_2" class="md-nav__link">
    Looking at the results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/thegeneschool/metagenomics/edit/master/docs/hic.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="exploiting-dna-dna-proximity">Exploiting DNA-DNA Proximity<a class="headerlink" href="#exploiting-dna-dna-proximity" title="Permanent link">&para;</a></h1>
<h2 id="what-is-metagenomic-hi-c">What is Metagenomic Hi-C?<a class="headerlink" href="#what-is-metagenomic-hi-c" title="Permanent link">&para;</a></h2>
<p>As the name suggests, Metagenomic Hi-C is an adaptation of the original Hi-C protocol -- used for the 3-dimensional study of chromosome structure  [1] -- to shotgun metagenome analysis.
Metagenomic Hi-C is a promising addition to our analysis toolbox, as it makes it possible for genome binning to be carried out accurately and precisely from a single sample rather than requiring large collection of samples (such as in timeseries binning).</p>
<p>The Hi-C protocol was designed to capture genome-wide evidence of DNA molecules in close physical proximity <em>in vivo</em>.
With the flexible nature of a chromosome and its capacity to bend back on itself, the most frequently observed interactions are usually those between nearby loci on the same chromosome (intra-chromosomal).
Though the rate of interaction between loci falls off quickly with separation, it does not go to zero and long-distance contacts within a chromosome remain observable.
Less frequent than intra-chromosomal interactions are those between separate molecules (chromosomes, plasmids, etc) within the same cell (inter-chromosomal).
To what degree we see inter-chromosomal interactions depends greatly on the organism(s).
The least common are inter-cellular interactions and are often seen at rates far below those found within a cell.</p>
<p>The proximity information captured in Hi-C sequencing data can be readily applied to improve draft genome and metagenome assemblies.
In metagenomics we use the proximity information to infer which assembly fragments (contigs) originated from the same cell (or chromosome).
Though the power of the "proximity signal" should ultimately make it possible to infer individual genotypes, current tools stop at the same resolution as timeseries binning, associating assembly fragments into MAGs.</p>
<h2 id="the-hi-c-library-protocol">The Hi-C Library Protocol<a class="headerlink" href="#the-hi-c-library-protocol" title="Permanent link">&para;</a></h2>
<div class="admonition info">
<p class="admonition-title">Major steps of the Hi-C protocol</p>
<p><img alt="zoomify" src="../img/hi-c_steps.png" style="margin-left: 10%; width:75%" /></p>
</div>
<h3 id="basic-concept">Basic Concept<a class="headerlink" href="#basic-concept" title="Permanent link">&para;</a></h3>
<p>Though Hi-C sequencing data is generated using conventional high-throughput Illumina paired-end sequencing, the more complicated Hi-C protocol produces a very different library.
Unlike traditional shotgun sequencing, where read pairs originate from nearby regions on the same contiguous DNA molecule (i.e. chromosome, plasmid, etc), the Hi-C protocol can generate read-pairs from any two strands of DNA that were in close physical proximity within the cell.
So long as the two DNA loci interact, regardless of their location with the genome, there is a chance of producing a Hi-C read-pair from that interaction.
Hi-C read pairs can therefore originate from far-away regions of the same molecule or from entirely different molecules.</p>
<p>To achieve this, the Hi-C library protocol involves several steps upstream of Illumina adapter ligation and is more challenging to execute in the lab.
As a result, commercial kits have appeared which aim simplify the process and improve quality and consistency of the resulting libraries.</p>
<h4 id="some-commercial-companies-offering-varieties-of-hi-c-kits-and-services">Some commercial companies offering varieties of Hi-C kits and services<a class="headerlink" href="#some-commercial-companies-offering-varieties-of-hi-c-kits-and-services" title="Permanent link">&para;</a></h4>
<ul>
<li><a href="https://www.phasegenomics.com/">Phase Genomics</a></li>
<li><a href="https://arimagenomics.com/">Arima Genomics</a></li>
<li><a href="https://dovetailgenomics.com">Dovetail Genomics</a></li>
</ul>
<h3 id="rough-protocol-outline">Rough Protocol Outline<a class="headerlink" href="#rough-protocol-outline" title="Permanent link">&para;</a></h3>
<ol>
<li>
<p><strong>DNA Fixation:</strong> Beginning with intact cells, the first step of the Hi-C protocol is formalin fixation. The act of cross-linking "locks in" close-by conformation arrangements within the DNA that existed in the cells at the time of fixation.</p>
</li>
<li>
<p><strong>Cell Lysis:</strong> The cells are lysed and DNA-protein complexes extracted and purified.</p>
</li>
<li>
<p><strong>Restriction Digest:</strong> The DNA is digested using a restriction endonuclease. In metagenomic Hi-C, enzymes with large overhangs and 4-nt recognition sites are common choices (i.e. Sau3AI, MluCI, DpnII). Differences in the GC bias of the recognition site and the target DNA is an important factor, as inefficient digestion will produce fewer Hi-C read-pairs.</p>
</li>
<li>
<p><strong>Biotin Tagging:</strong> The overhangs produced during digestion are end-filled with biotinylated nucleotides.</p>
</li>
<li>
<p><strong>Free-end Ligation:</strong> in dilute conditions or immobilised on a substrate, free-ends protruding from the DNA-protein complexes are ligated. This stochastic process favours any two ends which were nearby within the complex, however random ligation and self-ligation can occur and the reads resulting from such events create noise/error in the data.</p>
</li>
<li>
<p><strong>Crosslink Reversal:</strong> The bonds created by formalin fixation are removed, allowing the now free DNA to be purified. Proteinase digestion is common following this step.</p>
</li>
<li>
<p><strong>Un-ligated End Clean-up:</strong> Free-ends which failed to form ligation products are unwanted in subsequent steps but could still be biotin tagged. To minimise their inclusion, a light exonuclease 3' to 5' favoured chew-back can be applied.</p>
</li>
<li>
<p><strong>DNA Shearing:</strong> With ligation completed, the DNA is mechanically sheared and the size range of the resulting fragment selected suitability with Illumina paired-end sequencing. </p>
</li>
<li>
<p><strong>DNA repair:</strong> Sonication can lead to damaged ends and nicks, which can be repaired.</p>
</li>
<li>
<p><strong>Proximity Ligation Enrichment:</strong> Biotin tagged fragments are pulled down using affinity purification. </p>
</li>
<li>
<p><strong>Adapter Ligation:</strong> Illumina paired-end adapter ligation and associated steps to produce a sequencing library are now applied.</p>
</li>
</ol>
<h2 id="quality-control-is-my-library-ok">Quality Control: is my library ok?<a class="headerlink" href="#quality-control-is-my-library-ok" title="Permanent link">&para;</a></h2>
<p>Due to the highly variable nature of samples taken directly from an environment, the possible presence of enzyme inhibitors, and other reasons, the metagenomic Hi-C protocol may not always work with high efficiency, or at all.
Problems with sample processing can sometimes be identified prior to sequencing, for example if the measured library yield is very low.
But other problems may be difficult to diagnose prior to sequencing, for example, it is very difficult to estimate the efficiency of proximity ligation and thus the fraction of sequencing reads that will be Hi-C reads without actually sequencing the sample.
The reads that are not Hi-C reads are essentially conventional shotgun read-pairs, which in the context of Hi-C are uninformative.</p>
<p>The value of the proximity information contained within a Hi-C read-set is high enough that even with a low efficiency library, it can be worthwhile to compensate for low efficiency by sequencing more deeply to obtain more Hi-C read pairs.
Thus, determining the percentage of Hi-C read-pairs in the library is important, because it can tell us how deeply we need to sequence our library.
Rather than submit a library to a large and costly sequencing run immediately, a small pilot sequencing run can be sufficient to confidently estimate the Hi-C efficiency.
Armed with this information, a researcher can make informed decisions about further action; such as whether a candidate Hi-C library should be fully sequenced and to what depth.</p>
<p>We now discuss various ways to estimate the fraction of Hi-C reads in a library.</p>
<h4 id="evidence-of-proximity-ligation">Evidence of Proximity Ligation<a class="headerlink" href="#evidence-of-proximity-ligation" title="Permanent link">&para;</a></h4>
<h5 id="long-range-pairs">Long-range pairs<a class="headerlink" href="#long-range-pairs" title="Permanent link">&para;</a></h5>
<p>The separation distance for intra-chromosomal Hi-C read-pairs is bounded only by the length of the chromosome.
This is unlike shotgun read-pairs whose separation is rarely more than a thousand nucleotides due to chemistry limitations on Illumina instruments.
Though it is less direct than looking for evidence of cut-site duplication events, a simple count of the number of read pairs than map far apart can be used to infer the percentage of Hi-C read-pairs.
The main limitation of this approach is that it requires a high quality reference assembly against which the Hi-C read pairs can be mapped.
In many metagenomic projects, such an assembly is not actually available, either because a good quality shotgun metagenome dataset has not yet been generated for the sample or because there are no close reference genomes in public databases for the organisms in the sample.</p>
<div class="admonition success">
<p>Counting the number of of pairs which map at long-range can as be used to infer the percentage of Hi-C read-pairs.</p>
</div>
<h5 id="proximity-ligation-junctions">Proximity ligation junctions<a class="headerlink" href="#proximity-ligation-junctions" title="Permanent link">&para;</a></h5>
<p>In the Hi-C protocol outlined above, the steps of creating free-ends through enzymatic digestion, subsequent end-repair, and finally re-ligation introduces an artifact at the junction point.
This artifact is a short sequence duplication and the exact sequence enzyme dependent.
The duplication is produced when the cleavage site overhangs are subjected to end-repair.
When the repaired blunt ends are subsequently ligated, their junction contains two copies of the overhang sequence.</p>
<div class="admonition info">
<p class="admonition-title">Proximity junction for the enzyme Sau3AI</p>
<p>Sau3AI is a 4-cutter with a 4nt overhang.</p>
<p>Native DNA
<div class="codehilite"><pre><span></span>5`-XXXXGATCYYYY-3&#39;
3&#39;-xxxxCTAGyyyy-5&#39;
</pre></div>
Free-ends post cleavage have overhangs
<div class="codehilite"><pre><span></span>5`-XXXX     GATCYYYY-3&#39;
3&#39;-xxxxCTAG     yyyy-5&#39;
</pre></div>
Blunt ends after end-repair with biotinylated nucleotides
<div class="codehilite"><pre><span></span>5`-XXXXGATC GATCYYYY-3&#39;
3&#39;-xxxxCTAG CTAGyyyy-5&#39;
</pre></div>
Free-end ligation produces a duplication at the junction
<div class="codehilite"><pre><span></span>5`-XXXXGATCGATCYYYY-3&#39;
3&#39;-xxxxCTAGCTAGyyyy-5&#39;
</pre></div></p>
</div>
<p>After the steps of DNA shearing and size selection, fragments which eventually go on to DNA sequencing can contain a junction at any point along their extent.
An Illumina paired-end sequencing run, which generates reads at either end of a fragment, thus has two chances to read through the junction.
This approach has two main drawbacks:</p>
<ul>
<li>If the fragments in the sequencing library are long relative to the read length, there may be junctions within the fragment that remain unobserved, and this fact must be corrected for in the estimate.</li>
<li>If the sample was degraded there may have been free ends that were not created by an enzyme cut.
These could become proximity-ligated, and therefore ligation junctions may exist that do not contain the obvious junction sequence.</li>
</ul>
<div class="admonition success">
<p>Searching the read-set for examples of this junction sequence is one means of measuring the percentage of Hi-C read-pairs in a given library.</p>
</div>
<h2 id="using-a-qc-tool">Using a QC Tool<a class="headerlink" href="#using-a-qc-tool" title="Permanent link">&para;</a></h2>
<p>Though the methodology of Hi-C QC has yet to achieve standardisation, a number of tools exist which offer forms of QC testing.
These tools can be found as components of Hi-C analysis pipelines (<code>HiCpipe</code>), embedded in tool-suites (<code>HiCExplorer</code>) and as stand-alone command-line tools (<code>hic_qc</code>, <code>qc3C</code>) [2-5].</p>
<p>In this tutorial we will use <code>qc3C</code> to assess Hi-C library quality in two different ways: </p>
<ol>
<li>Read mapping based analysis</li>
<li><em>k</em>-mer based analysis. </li>
</ol>
<h3 id="get-some-hi-c-data-and-tools">Get some Hi-C data and tools<a class="headerlink" href="#get-some-hi-c-data-and-tools" title="Permanent link">&para;</a></h3>
<p>For a single timepoint, we'll first download a small Hi-C read-set and its associated shotgun read-set.
Also, we'll pull down the Docker images for <code>qc3C</code> (quality testing) and <code>bin3C</code> (metagenome binning)</p>
<div class="admonition example">
<p class="admonition-title">Get some Hi-C data and the qc3C and bin3C docker images</p>
<div class="codehilite"><pre><span></span><span class="c1"># download the read-set</span>
parallel-fastq-dump/parallel-fastq-dump -s SRR9323809 -s SRR8960211 --threads <span class="m">4</span> --outdir hic_data/ <span class="se">\</span>
    --minSpotId <span class="m">0</span> --maxSpotId <span class="m">1000000</span> --split-files --gzip
<span class="c1"># fetch the qc3C image</span>
sudo docker pull cerebis/qc3c:alpine
<span class="c1"># fetch the bin3c image</span>
sudo docker pull cerebis/bin3c:latest
</pre></div>

</div>
<h3 id="create-a-metagenome-assembly-and-map-hi-c-reads">Create a metagenome assembly and map Hi-C reads<a class="headerlink" href="#create-a-metagenome-assembly-and-map-hi-c-reads" title="Permanent link">&para;</a></h3>
<p>The shotgun dataset has been limited to 1M read-pairs, this is only 1/165<sup>th</sup> of the total.
Obviously this reduction in total coverage will lead to a much sparser sampling of the community and a more fragmented assembly.
Nevertheless, it is enough data to demonstrate the concept of QC and Hi-C metagenome binning.</p>
<p>Lets assemble the shotgun data and map both library types to the resulting contigs.</p>
<div class="admonition example">
<p class="admonition-title">Create a SPAdes assembly</p>
<div class="codehilite"><pre><span></span><span class="c1"># enter the hic data directory</span>
<span class="nb">cd</span> hic_data
<span class="c1"># launch spades in metagenomic mode</span>
docker run -v <span class="nv">$PWD</span>:/opt/app-root cerebis/bin3c:latest spades.py --meta -1 SRR8960211_1.fastq.gz -2 SRR8960211_2.fastq.gz -o asm
</pre></div>

</div>
<p>Once we have our assembly, we can map the Hi-C reads to the contigs. This mapping forms the basis of what binning tools like <code>bin3C</code> use to infer proximity associations between DNA loci.</p>
<div class="admonition example">
<p class="admonition-title">Map Hi-C reads to assembly contigs</p>
<div class="codehilite"><pre><span></span><span class="c1"># index the contig fasta</span>
sudo docker run -v <span class="nv">$PWD</span>:/opt/app-root cerebis/bin3c:latest bwa index asm/contigs.fasta
<span class="c1"># map hi-c reads to the contigs </span>
docker run -v <span class="nv">$PWD</span>:/opt/app-root cerebis/bin3c:latest /bin/bash -c <span class="s2">&quot;bwa mem -t4 -5SP asm/contigs.fasta SRR9323809_1.fastq.gz SRR9323809_2.fastq.gz | samtools view -uS -F 0x904 - | samtools sort -@4 -n -o hic_to_ctg.bam -&quot;</span>
<span class="c1"># map shotgun reads to the contigs </span>
docker run -v <span class="nv">$PWD</span>:/opt/app-root cerebis/bin3c:latest /bin/bash -c <span class="s2">&quot;bwa mem -t4 -5SP asm/contigs.fasta SRR8960211_1.fastq.gz SRR8960211_2.fastq.gz | samtools view -uS -F 0x904 - | samtools sort -@4 -n -o wgs_to_ctg.bam -&quot;</span>
<span class="c1"># return to your home directory</span>
<span class="nb">cd</span> 
</pre></div>

</div>
<div class="admonition tip">
<p class="admonition-title">Breaking down the read mapping command</p>
<ol>
<li>We are providing Docker access to our host filesystem using a bind mount (<code>-v $PWD:/opt/app-root</code>)</li>
<li>Next we are specifying the image to run (<code>cerebis/bin3c:latest</code>).</li>
<li>Finally we include the actual call, which involves pipes between <code>bwa</code> and <code>samtools</code>. To achieve this we pass the <code>bash</code> shell our complex command, which it then executes within the container. </li>
</ol>
</div>
<h3 id="bam-mode-analysis-with-qc3c">BAM mode analysis with qc3C<a class="headerlink" href="#bam-mode-analysis-with-qc3c" title="Permanent link">&para;</a></h3>
<p>We begin with the conceptually simpler approach to Hi-C QC, counting the number of long-range read-pairs from a Hi-C read-set.
As a rule of thumb, to work well with Illumina sequencing, the physical size of fragments is likely to be &lt;1000nt.
We shall therefore simply count the number of pairs which map with a separation &gt;1000, &gt;5000 and &gt;10,000 nt.</p>
<p>As the mapped read-pair separation grows beyond 1000nt we expect an increasingly large proportion of these pairs will be due to Hi-C proximity ligation.</p>
<p>Conversely, when analysing a shotgun read-set, we expect to see very few.</p>
<div class="admonition example">
<p class="admonition-title">Run a BAM based analysis</p>
<div class="codehilite"><pre><span></span><span class="c1"># enter the hic data folder</span>
<span class="nb">cd</span> hic_data
<span class="c1"># perform bam based QC analysis on Hi-C data </span>
docker run -v <span class="nv">$PWD</span>:/opt/app-root cerebis/qc3c:alpine ash -c <span class="s2">&quot;qc3C bam -m 400 -e Sau3AI -e MluCI -b hic_to_ctg.bam&quot;</span>
<span class="c1"># now perform the analysis on shotgun data</span>
docker run -v <span class="nv">$PWD</span>:/opt/app-root cerebis/qc3c:alpine ash -c <span class="s2">&quot;qc3C bam -m 400 -e Sau3AI -e MluCI -b wgs_to_ctg.bam&quot;</span>
<span class="c1"># return to your home directory</span>
<span class="nb">cd</span> 
</pre></div>

</div>
<h4 id="estimation-of-fragment-size">Estimation of fragment size<a class="headerlink" href="#estimation-of-fragment-size" title="Permanent link">&para;</a></h4>
<p>As we are not sure of the mean insert length, we guess it to be 400nt.
This is one handy benefit of the BAM mode approach, we can infer a reasonably good estimate of the fragment (insert) length.
In <em>k</em>-mer mode, having an accurate value becomes important if we wish to estimate the probable fraction of unobserved proximity junctions.
That is to say, for long fragments and short reads, much of each fragment goes unsequenced and so we will miss the evidence of junctions.</p>
<p>In our resulting log from above, <code>qc3C</code> reports an observed mean fragment length of 445nt for the Hi-C read-set.</p>
<div class="admonition note">
<div class="codehilite"><pre><span></span>INFO     | 2019-06-23 18:55:18,652 | qc3C.bam_based | Observed mean of short-range pair separation: 445nt
</pre></div>

</div>
<h4 id="looking-at-the-results">Looking at the results<a class="headerlink" href="#looking-at-the-results" title="Permanent link">&para;</a></h4>
<p>The last three lines of the output from <code>qc3C</code> report the absolute number and relative fraction of read-pairs that mapped far apart.</p>
<p>For the Hi-C library, the fraction of pairs separated by more than 10kb was ~2.6%, while in contrast for the known shotgun library the fraction was &lt;0.006%.</p>
<div class="admonition note">
<p>Hi-C result
<div class="codehilite"><pre><span></span>INFO     | 2019-11-19 08:31:13,975 | qc3C.bam_based | Long-range distance intervals:    1000nt,    5000nt,   10000nt
INFO     | 2019-11-19 08:31:13,975 | qc3C.bam_based | Number of observed pairs:          366,       169,        84
INFO     | 2019-11-19 08:31:13,975 | qc3C.bam_based | Relative fraction of all cis:   0.1149,   0.05308,   0.02638
INFO     | 2019-11-19 08:31:13,976 | qc3C.bam_based | Relative fraction of all pairs: 0.04934,   0.02278,   0.01132
</pre></div></p>
<p>Shotgun result
<div class="codehilite"><pre><span></span>INFO     | 2019-11-19 08:31:42,171 | qc3C.bam_based | Long-range distance intervals:    1000nt,    5000nt,   10000nt
INFO     | 2019-11-19 08:31:42,171 | qc3C.bam_based | Number of observed pairs:          284,        43,        23
INFO     | 2019-11-19 08:31:42,172 | qc3C.bam_based | Relative fraction of all cis:   0.0007268,   0.0001101,   5.886e-05
INFO     | 2019-11-19 08:31:42,172 | qc3C.bam_based | Relative fraction of all pairs: 0.0003572,   5.408e-05,   2.893e-05
</pre></div></p>
</div>
<p>It is important to keep in mind that as the assembly was made from a shotgun dataset downsampled to very low coverage, the length of contigs is significantly lower than would have been obtained with the full dataset.
This will impair the odds of seeing pairs which map far away.
Despite this, the 98% of pairs in the shotgun library map to the same contig sequence.
In contrast, 73% of Hi-C pairs mapped to the same contig.</p>
<p>Overall, for BAM mode, the reliability of results is connected to the quality and completeness of the available references.</p>
<div class="admonition tip">
<p class="admonition-title">Breaking down the invocation of <code>qc3C</code></p>
<ol>
<li>We are providing Docker access to our host filesystem using a bind mount (<code>-v $PWD:/opt/app-root</code>)</li>
<li>Next we are specifying the image to run (<code>cerebis/qc3c:alpine</code>).</li>
<li>Finally we include the actual call to <code>qc3C</code>, along with its own options. </li>
</ol>
</div>
<div class="admonition info">
<p class="admonition-title">Potential Concerns</p>
<ul>
<li>Requires a reference sequence:<ul>
<li>a completed genome</li>
<li>a draft assembly.</li>
</ul>
</li>
<li>May not be an option when:<ul>
<li>no reference is available</li>
<li>only a very highly fragmented draft exists </li>
</ul>
</li>
<li>Computational requirements:<ul>
<li>create the shotgun assembly</li>
<li>map Hi-C reads to the reference </li>
</ul>
</li>
<li>The distribution of assembly contig lengths will affect the maximum mapping distance.</li>
</ul>
</div>
<h3 id="k-mer-mode-analysis-with-qc3c"><em>k</em>-mer mode analysis with qc3C<a class="headerlink" href="#k-mer-mode-analysis-with-qc3c" title="Permanent link">&para;</a></h3>
<p>To analyze our small Hi-C read-set using a <em>k</em>-mer approach, we will first create a <em>k</em>-mer library using Jellyfish.
Since this Hi-C experiment used the 4-cutter Sau3AI, which produces an 8-mer junction, we'll use a mer size of 24 for the library.
This will give us 8 nucleotides either side of any prospective junction, for specificity.</p>
<p>Choosing how large a <em>k</em>-mer size to use for the library is a trade-off between computational complexity and minimising false discovery.
You can try sizes bigger and smaller.</p>
<div class="admonition example">
<p class="admonition-title">Running k-mer based analyses</p>
<p>On the Hi-C data
<div class="codehilite"><pre><span></span><span class="c1"># enter the hic data folder</span>
<span class="nb">cd</span> hic_data
<span class="c1"># create a generator used by Jellyfish for compressed files</span>
rm -f gen_fq <span class="o">&amp;&amp;</span> <span class="k">for</span> fn in <span class="sb">`</span>ls SRR9323809*fastq.gz<span class="sb">`</span><span class="p">;</span> <span class="k">do</span> <span class="nb">echo</span> gzip -dc <span class="nv">$fn</span> &gt;&gt; gen_fq<span class="p">;</span> <span class="k">done</span>
<span class="c1"># use jellyfish to create a 24-mer library from FastQ reads</span>
docker run -v <span class="nv">$PWD</span>:/opt/app-root cerebis/qc3c:alpine jellyfish count -t <span class="m">4</span> -m <span class="m">24</span> -s 100M -C -g gen_fq -o hic24.jf
<span class="c1"># perform k-mer based QC analysis</span>
docker run -v <span class="nv">$PWD</span>:/opt/app-root cerebis/qc3c:alpine ash -c <span class="s2">&quot;qc3C kmer -m 445 -s 1234 -e Sau3AI -l hic24.jf -r SRR9323809_1.fastq.gz -r SRR9323809_2.fastq.gz&quot;</span>
<span class="c1"># return back to your home directory</span>
<span class="nb">cd</span> 
</pre></div>
Now on the shotgun data
<div class="codehilite"><pre><span></span><span class="c1"># enter the hic data folder</span>
<span class="nb">cd</span> hic_data
<span class="c1"># create a generator used by Jellyfish for compressed files</span>
<span class="nb">echo</span> <span class="s2">&quot;gzip -dc SRR8960211_1.fastq.gz&quot;</span> &gt; gen_sg
<span class="c1"># use jellyfish to create a 24-mer library from FastQ reads</span>
docker run -v <span class="nv">$PWD</span>:/opt/app-root cerebis/qc3c:alpine jellyfish count -t <span class="m">4</span> -m <span class="m">24</span> -s 100M -C -g gen_sg -o wgs24.jf
<span class="c1"># perform k-mer based QC analysis</span>
docker run -v <span class="nv">$PWD</span>:/opt/app-root cerebis/qc3c:alpine ash -c <span class="s2">&quot;qc3C kmer -m 306 -s 1234 -e Sau3AI -l wgs24.jf -r SRR8960211_1.fastq.gz -p 0.05&quot;</span>
<span class="c1"># return back to your home directory</span>
<span class="nb">cd</span> 
</pre></div></p>
</div>
<h4 id="looking-at-the-results_1">Looking at the results<a class="headerlink" href="#looking-at-the-results_1" title="Permanent link">&para;</a></h4>
<p>Looking at the inferred fraction of true Hi-C read-pairs, we can immediately see that the low coverage shotgun assembly appears to have drastically skewed the estimate of signal (2.7% vs 5.8%). 
When estimating Hi-C content using an assembly based method, we require a decent assembly.</p>
<p>After adjustment for the unobserved fraction (0.32), the k-mer based estimate of signal is increased to 7.6%.</p>
<p>In contrast, for the conventional shotgun data-set we should expect an estimate near zero and we see a fairly low value of 0.094%. This tells us that were we to see such low values for a Hi-C library, something has gone seriously wrong with the library prep and further sequencing would be best abandoned.</p>
<p>As another check, if we look further up the log we can see that the number of observed junctions was 0.098%, while we could expect 0.0015% purely by random chance.
Although this difference might appear significant, it is put clear in perspective when compared to the Hi-C result, where 5.9% of reads contained the junction sequence.</p>
<p>This Hi-C library appears to have a reasonable fraction of Hi-C reads.
If it were a pilot sequencing run, this result would be sufficient evidence to commit to a larger sequencing run.</p>
<div class="admonition example">
<p>Hi-C result
<div class="codehilite"><pre><span></span>INFO     | 2019-11-19 08:35:53,320 | qc3C.kmer_based | Number of unique observations used in empirical p-value: wgs 348,653, hic 21,592
INFO     | 2019-11-19 08:35:53,470 | qc3C.kmer_based | Estimated Hi-C read fraction via p-value sum method: 5.753 ± 0.004587 %
INFO     | 2019-11-19 08:35:53,470 | qc3C.kmer_based | Observed mean read length for paired reads: 151nt
INFO     | 2019-11-19 08:35:53,470 | qc3C.kmer_based | For supplied insert length of 445nt, estimated unobserved fraction: 0.3213
INFO     | 2019-11-19 08:35:53,470 | qc3C.kmer_based | Adjusted estimation of Hi-C read fraction: 7.601 ± 0.006062 %
</pre></div></p>
<p>Shotgun result
<div class="codehilite"><pre><span></span>INFO     | 2019-11-19 08:37:46,411 | qc3C.kmer_based | Number of unique observations used in empirical p-value: wgs 49,715, hic 49
INFO     | 2019-11-19 08:37:46,433 | qc3C.kmer_based | Estimated Hi-C read fraction via p-value sum method: 0.09371 ± 0.003008 %
INFO     | 2019-11-19 08:37:46,434 | qc3C.kmer_based | Observed mean read length for paired reads: 151nt
INFO     | 2019-11-19 08:37:46,434 | qc3C.kmer_based | For supplied insert length of 306nt, estimated unobserved fraction: 0.01307
INFO     | 2019-11-19 08:37:46,434 | qc3C.kmer_based | Adjusted estimation of Hi-C read fraction: 0.09493 ± 0.003048 %
</pre></div></p>
</div>
<div class="admonition tip">
<p class="admonition-title">Breaking down the invocation of <code>jellyfish</code></p>
<ol>
<li>We are providing Docker access to our host filesystem using a bind mount (<code>-v $PWD:/opt/app-root</code>)</li>
<li>Next we are specifying the image to run (<code>cerebis/qc3c:alpine</code>).</li>
<li>Finally we include the actual call to <code>jellyfish</code>, along with its own options. </li>
</ol>
</div>
<div class="admonition info">
<p class="admonition-title">K-mer mode requirements</p>
<ul>
<li>No-assembly or reference required</li>
<li>Requires making a k-mer library from FastQ reads</li>
<li>Analyse FastQ reads against library</li>
</ul>
</div>
<h2 id="metagenome-binning-with-hi-c-data">Metagenome Binning with Hi-C Data<a class="headerlink" href="#metagenome-binning-with-hi-c-data" title="Permanent link">&para;</a></h2>
<p>As we saw by processing timeseries data with <code>MetaBAT2</code>, metagenome binning can also be performed using Hi-C.
The primary difference here is that accurate and precise binning is possible from a single timepoint, so long as you have you have both a shotgun and Hi-C read-set.
Currenty, only a few tools exist which were made specifically for this purpose (<code>bin3C</code>, <code>ProxiMeta</code>) [6,7].</p>
<p>In this tutorial we will use bin3C, as it is open-source and easy to use.</p>
<p>With a set of reference sequences and Hi-C to reference BAM file in hand (from above), a <code>bin3C</code> analysis has two stages: first create the contact map, second cluster the contact map.
We also need to know what enzyme(s) was used in generating the library.
It is becoming more common to generate libraries from two enzymes with differing GC biases to improve Hi-C coverage within a community.
In such cases, we simply specify them when creating the map, as here.</p>
<div class="admonition example">
<p class="admonition-title">Perform Hi-C metagenome binning</p>
<div class="codehilite"><pre><span></span><span class="c1"># enter hic data directory</span>
<span class="nb">cd</span> hic_data
<span class="c1"># Make contact map</span>
docker run -v <span class="nv">$PWD</span>:/opt/app-root cerebis/bin3c:latest bin3C mkmap -v --clobber -e Sau3AI -e MluCI --eta --bin-size <span class="m">5000</span> asm/contigs.fasta hic_to_ctg.bam bin3C_out
<span class="c1"># Cluster contact map</span>
docker run -v <span class="nv">$PWD</span>:/opt/app-root cerebis/bin3c:latest bin3C cluster -v --clobber --only-large bin3C_out/contact_map.p.gz bin3C_out 
<span class="c1"># return to your home directory</span>
<span class="nb">cd</span>
</pre></div>

</div>
<h4 id="looking-at-the-results_2">Looking at the results<a class="headerlink" href="#looking-at-the-results_2" title="Permanent link">&para;</a></h4>
<div class="admonition warning">
<p class="admonition-title">Our example data-set was insufficient</p>
<p>Unfortunately, the example data we have used here is insufficient to get a useful result from bin3C. Neither the shotgun, nor the Hi-C data-set was sequenced deeply enough. 
In reality, we artificially limited the depth when downloading the FastQ data with <code>--max-spot</code>. 
Were we to go back and obtain the full data-sets, we would achieve a much better result (see the figure below).</p>
</div>
<p>After <code>bin3C</code> completes the clustering step, the three outputs which are possibly the most interesting are:</p>
<ol>
<li>The comma delimited <code>cluster_report.csv</code> table<ul>
<li>the cluster_report details characteristics of each identified cluster.</li>
<li>An id, name, number of contigs, total extent, mean GC and coverage, etc.</li>
<li>Mean coverage is a reasonable gauge of relative abundance</li>
</ul>
</li>
<li>A qualitative overview of the result in the form of a heatmap <code>cluster_plot.png</code><ul>
<li>contigs are organized by cluster, followed by descending length</li>
<li>a pixel represents a 5kb bin and therefore contigs are represented proportional to their length.</li>
</ul>
</li>
<li>Clustered sequence data<ul>
<li>the <code>fasta/</code> directory contains cluster FastA data</li>
</ul>
</li>
</ol>
<p>The cluster plot is a visual representation of the clustered contact map.
As a metagenomic assembly can easily contain 50k to 100k contigs, which as a NxN image would be very large, the plot's resolution is constrained by default to 4000x4000 pixels to conserve memory.</p>
<p>Once organised, a good Hi-C read-set will have produced a map that is clearly in block-diagonal form with high contrast.
Also, a significant proportion of the map's field is should be dark, reflecting the fact that inter-cellular interactions are rare.
In cases where there was insufficient signal, there will only be a few (if any) large clusters and they will look sparse internally.</p>
<div class="admonition info">
<p class="admonition-title">Example cluster plot of a human microbome</p>
<p><img alt="" src="../img/cluster_plot.png" style="margin-left: 12%; width:75%" /></p>
<p><strong>Right click to get a larger image</strong></p>
<p>MAG clusters of a single-timepoint human faecal microbiome (PRJNA413092) as determined by <code>bin3C</code>.
The shotgun read-set consisted of 249M pairs, while the Hi-C library contained 41M pairs (~10% Hi-C).
Of the identified MAGs, 55 where &gt;90% complete, &lt;5% contamination as estimated by CheckM [6]. </p>
</div>
<p>If you are familiar with the appearance of a single-genome microbial contact map, at sufficient zoom you may notice that the individual clusters do not look the same.
This is due to the fact that the proper ordering of contigs (scaffolding) within the clusters has not been established.</p>
<p>After reaching this point, standard downstream analyses would be to profile the taxonomic composition of the clusters with tools such as <a href="https://github.com/Ecogenomics/CheckM"><code>CheckM</code></a>.</p>
<div class="admonition info">
<p class="admonition-title">Expectations from Hi-C binning</p>
<p><img alt="" src="../img/ami_wgs_vs_hic.png" style="margin-left: 25%; width:50%" /></p>
<p>Hi-C metagenome binning is dependent on both the quality of the shotgun assembly and the depth of true Hi-C read-pairs.
For a Hi-C data-set with strong signal (perhaps 10%) and reasonable depth of sequencing (100M pairs), it is reasonable to expect 50-100 nearly complete MAGs [6].</p>
</div>
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<p>1.
Lieberman-Aiden, E., van Berkum, N. L., Williams, L., Imakaev, M., Ragoczy, T., Telling, A., … Dekker, J. (2009). Comprehensive mapping of long-range interactions reveals folding principles of the human genome. <em>Science</em>, 326(5950), 289–293. <a href="https://doi.org/10.1126/science.1181369">https://doi.org/10.1126/science.1181369</a></p>
<ol>
<li>
<p>fchen. (n.d.). HiCpipe. Retrieved from <a href="https://github.com/ChenFengling/HiCpipe">https://github.com/ChenFengling/HiCpipe</a></p>
</li>
<li>
<p>Wolff, J., Bhardwaj, V., Nothjunge, S., Richard, G., Renschler, G., Gilsbach, R., … Grüning, B. A. (2018). Galaxy HiCExplorer: a web server for reproducible Hi-C data analysis, quality control and visualization. <em>Nucleic Acids Research</em>, 46(W1), W11–W16. <a href="https://doi.org/10.1093/nar/gky504">https://doi.org/10.1093/nar/gky504</a></p>
</li>
<li>
<p>DeMaere, M. Z., Darling A. E. (n.d.). qc3C. Retrieved from <a href="https://github.com/cerebis/qc3C">https://github.com/cerebis/qc3C</a></p>
</li>
<li>
<p>hic_qc. (n.d.). Retrieved from <a href="https://github.com/phasegenomics/hic_qc">https://github.com/phasegenomics/hic_qc</a></p>
</li>
<li>
<p>DeMaere, M. Z., &amp; Darling, A. E. (2019). bin3C: exploiting Hi-C sequencing data to accurately resolve metagenome-assembled genomes. <em>Genome Biology</em>, 20(1), 46. <a href="https://doi.org/10.1186/s13059-019-1643-1">https://doi.org/10.1186/s13059-019-1643-1</a></p>
</li>
<li>
<p>Press, M. O., Wiser, A. H., Kronenberg, Z. N., Langford, K. W., Shakya, M., Lo, C.-C., … Liachko, I. (2017). Hi-C deconvolution of a human gut microbiome yields high-quality draft genomes and reveals plasmid-genome interactions (p. 198713). <a href="https://doi.org/10.1101/198713">https://doi.org/10.1101/198713</a></p>
</li>
</ol>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../binning/" title="Binning" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Binning
              </span>
            </div>
          </a>
        
        
          <a href="../anvio/" title="Visualization" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Visualization
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.ac79c3b0.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS_CHTML"></script>
      
    
  </body>
</html>